[LBSE] Fix dynamic updating of lighting-color/flood-color/flood-opacity

rwlbuis opened the pull request:
#### 0eb86e157166d4f1bd0740f78b445747d02877e6
<pre>
[LBSE] Fix dynamic updating of lighting-color/flood-color/flood-opacity
<a href="https://bugs.webkit.org/show_bug.cgi?id=275824">https://bugs.webkit.org/show_bug.cgi?id=275824</a>

Reviewed by Nikolas Zimmermann and Said Abou-Hallawa.

Fix dynamic updating of lighting-color/flood-color/flood-opacity in the same way
as legacy. In particular:
- flood-color/flood-opacity style changes for feFlood and feDropShadow
- lighting-color style changes for feDiffuseLighting and feSpecularLighting.

* LayoutTests/platform/mac-sonoma-wk2-lbse-text/TestExpectations:
* Source/WebCore/rendering/svg/RenderSVGResourceFilterPrimitive.cpp:
(WebCore::RenderSVGResourceFilterPrimitive::filterPrimitiveElement const):
(WebCore::RenderSVGResourceFilterPrimitive::styleDidChange):
* Source/WebCore/rendering/svg/RenderSVGResourceFilterPrimitive.h:

Canonical link: <a href="https://commits.webkit.org/280328@main">https://commits.webkit.org/280328@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6765d3355a8481c85b19459c7dad39b285b1aaad

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56309 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35635 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8781 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/59916 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6745 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/58435 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43257 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6939 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45601 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4700 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58338 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33516 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48583 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26471 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30295 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5913 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/5749 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/52297 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6185 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/61600 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/217 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6311 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/52870 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/217 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48649 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/52751 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12468 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/194 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31462 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32548 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33631 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32295 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann commented on Source/WebCore/rendering/svg/RenderSVGResourceFilterPrimitive.cpp:
```diff
@@ -61,4 +67,28 @@ void RenderSVGResourceFilterPrimitive::markFilterEffectForRebuild()
+        return;
+
+    Ref newStyle = style().svgStyle();
+    if (is<SVGFEFloodElement>(filterPrimitiveElement()) || is<SVGFEDropShadowElement>(filterPrimitiveElement())) {
```
I'd turn these into early returns, instead of `else if` constructs for readability.

rwlbuis replied:
Agreed, reads a bit better, fixed.

nikolaszimmermann reviewed (approved):
Looks great, thanks @rwlbuis.

rwlbuis committed:
[LBSE] Fix dynamic updating of lighting-color/flood-color/flood-opacity
https://bugs.webkit.org/show_bug.cgi?id=275824

Reviewed by Nikolas Zimmermann and Said Abou-Hallawa.

Fix dynamic updating of lighting-color/flood-color/flood-opacity in the same way
as legacy. In particular:
- flood-color/flood-opacity style changes for feFlood and feDropShadow
- lighting-color style changes for feDiffuseLighting and feSpecularLighting.

* LayoutTests/platform/mac-sonoma-wk2-lbse-text/TestExpectations:
* Source/WebCore/rendering/svg/RenderSVGResourceFilterPrimitive.cpp:
(WebCore::RenderSVGResourceFilterPrimitive::filterPrimitiveElement const):
(WebCore::RenderSVGResourceFilterPrimitive::styleDidChange):
* Source/WebCore/rendering/svg/RenderSVGResourceFilterPrimitive.h:

Canonical link: https://commits.webkit.org/280328@main
